cache/lru/cache.go

```diff
@@ -55,15 +71,26 @@ func (c *Cache[K, V]) Get(key K) (V, bool) {
 func (c *Cache[K, _]) Evict(key K) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
+	c.evict(key)
+}
 
+func (c *Cache[K, V]) evict(key K) {
+	value, ok := c.elements.Get(key)
+	if !ok {
+		return
+	}
+	c.onEvict(key, value)
 	c.elements.Delete(key)
 }
 
 func (c *Cache[_, _]) Flush() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	c.elements.Clear()
+	iter := c.elements.NewIterator()
+	for iter.Next() {
```

Comment:
nit: more pedantic nits...
```suggestion
	for iter := c.elements.NewIterator(); iter.Next();  {
```